backend/packages/Upgrade/src/api/services/ExperimentService.ts

```diff
@@ -1392,6 +1370,63 @@ export class ExperimentService {
         logger.error(error);
         throw error;
       }
+
+      // creating condition payload docs after getting conditionDocs and decisionPointDocs
+      if (!conditionPayloads) {
+        experiment = { ...experiment, conditionPayloads: [] };
+      }
+      const conditionPayloadDocToSave: Array<any> =
+        (conditionPayloads &&
+          conditionPayloads.length > 0 &&
+          conditionPayloads.map((conditionPayload) => {
+            const parentCondition = conditionPayload.parentCondition
```

Comment:
ternary isn't needed, the `if` side by itself is sufficient, will be `undefined` if `find` finds nothing, and if `conditionPayload.parentCondition` is undefined, `conditionIdMap.get` will return undefined also.

```typescript
const parentCondition = conditionDocs.find((doc) => doc.id ===  conditionIdMap.get(conditionPayload.parentCondition))
```

Reply:
ok